frontend/src/pug/_coupon_id.pug

```diff
@@ -4,42 +4,127 @@
+        |     %{ of Nothing }
+        |     %{ endcase }
+
+        |     %{ case couponGiftMinimumPrice coupon }
+        |     %{ of Just giftMinimumPrice }
+        span.highlightedBlock-main \#{giftMinimumPrice}円
+        |     %{ of Nothing }
+        |     %{ endcase }
+
+        |     %{ case couponGiftReferencePrice coupon }
+        |     %{ of Just giftReferencePrice }
+        span.highlightedBlock-main \#{giftReferencePrice}円
+        |     %{ of Nothing }
+        |     %{ endcase }
+
+        | %{ of CouponTypeSet }
+        |     %{ case couponSetContent coupon }
+        |     %{ of Just setContent }
+        span.highlightedBlock-sub \#{setContent}
+        |     %{ of Nothing }
+        |     %{ endcase }
+
+        |     %{ case couponSetPrice coupon }
+        |     %{ of Just setPrice }
+        span.highlightedBlock-main \#{setPrice}円
+        |     %{ of Nothing }
+        |     %{ endcase }
+
+        |     %{ case couponSetReferencePrice coupon }
+        |     %{ of Just setReferencePrice }
+        span.highlightedBlock-main \#{setReferencePrice}円
+        |     %{ of Nothing }
+        |     %{ endcase }
+
+        | %{ of CouponTypeOther }
+        |     %{ case couponOtherContent coupon }
+        |     %{ of Just otherContent }
+        span.highlightedBlock-sub \#{otherContent}
+        |     %{ of Nothing }
+        |     %{ endcase }
+
+        | %{ endcase }
+
+
       .card_body_expiration
         span.card_body_expiration-title
           | 有効期限
         span.card_body_expiration-body
-          | 2016年4月15日から 2016年4月30日まで
+          | %{ case couponValidFrom coupon }
+          | %{ of Just validFrom }
+          | \#{tshow validFrom}から
+          | %{ of Nothing }
+          | %{ endcase }
+
+          | %{ case couponValidUntil coupon }
+          | %{ of Just validUntil }
+          | \#{tshow validUntil}まで
+          | %{ of Nothing }
+          | %{ endcase }
   .annotation ※ この画面を店舗でご提示ください
   .location
     .location_map(data-latitude="41.40243" data-longitude="2.17551")
       iframe.location_map-frame(src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d6483.579410111866!2d139.6998074!3d35.6575525!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x60188b59d79dd993%3A0xdef2b8ece045887e!2z5LiD6Lyq54S86IKJIOWuieWuiSDmuIvosLflupc!5e0!3m2!1sja!2sjp!4v1471871614830" frameborder="0" style="border:0" allowfullscreen)
   .aboutStore
-    a.btn.outerBtn(href="/store") 七輪焼き肉・安安
+    a.btn.outerBtn(href="/store") \#{maybe "(no title)" (storeName . entityVal) maybeStoreEntity}
   .moreContents
-    p.moreContents_txt 2.5時間飲放題付きの料理4品です。
-    p.moreContents_txt 1. 前菜 2. 焼き肉盛り合わせ 3. 冷麺 4. アイスクリーム
-    p.moreContents_txt ※ 1グループ1回のご利用時の利用枚数制限はありません。
-    p.moreContents_txt ※ 21時以降のご予約のお客様が対象です。
+    | %{ case couponCouponType coupon }
+    | %{ of CouponTypeDiscount }
+    |     %{ case couponDiscountOtherConditions coupon }
+    |     %{ of Just discountOtherConditions }
+    p.moreContents_txt \#{discountOtherConditions}
```

Comment:
I guess this causes multi-line shown in one-line.
It should be rewritten with `forall` control statement such that:
```
| %{ forall condition <- lines discountOtherConditions }
p.moreContents_txt \#{ condition }
| %{ endforall }
```